interview-questions.md

```diff
@@ -7,53 +7,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. Rails follows an MVC pattern. What does that mean?
 
-  Your answer:
+  Your answer: From the little I understand and what I believe it is, it allows us to ruby in a certain order. Models view controller
 
-  Researched answer:
+  Researched answer: a pattern used to decouple user-interface (view), data(model), and application logic (controller). this pattern helps to achieve separation of concerns
+
+  *model, view, controller, framework that gives structure and organization to your code, architecture
 
 
 
 2. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: relational databases stores data that are related to each other
+
+  Researched answer: types of database that stores and provides access to data points that are related to one another
 
-  Researched answer:
+  *database with rows and columns, columns hold attributes of the data and rows are instances, non-relational database NOSQL (not only sql)
 
 
 
 3. What is object-relational mapping?
 
-  Your answer:
+  Your answer: from what I know, orm allows us to convert data that are not compatible
 
-  Researched answer:
+  Researched answer: a programming technique for converting data between incompatible type systems using object-oriented programming languages
 
+  *convert data between two imcompatible languages, ORM - Active Record
 
 
 4. What is a gem?
 
-  Your answer:
+  Your answer: when you are creating a ruby file, gem comes along with. Gem is a package manager for ruby
 
-  Researched answer:
+  Researched answer: is a package manager for the Ruby programming language that provides a standard format for distributing Ruby programs and libraries
 
+  *ruby package manager, iterate code 
 
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: To be honest, I need to familiarize myself with this one
+
+  Researched answer: a special relational database table column designated to uniquely identify each table record
```

Comment:
I appreciate not knowing something and being honest. Remember the guidance on interviews is: if you can't answer the question try answering a simpler version of the question. 

That and the research answer is here for you to explore and deepen your knowledge on the subject matter. So even writing a sentence or two on the Who, What, When, Where, Why, or How-- will greatly increase understanding and retention.